Module-6-Assignment/StudentSearchTest.cls

```diff
@@ -0,0 +1,59 @@
+/**
+ * Test class to test StudentSearch class methods.
+ */
+@isTest
+public class StudentSearchTest 
+{
+   /**
+    * Test method to get the student name in a particular class.
+    */ 
+   @isTest
+   public static void getNameOfOneStudent()
+   {
+       Class__c cls = new Class__c(Name='x',MaxSize__c=2);
+       insert cls;
+       Student__c relatedStudents = new Student__c(Name='xyz', Class__c=cls.Id, Sex__c='Female');
+       insert relatedStudents;
+       List<Student__c> namesOfStudents = StudentSearch.searchStudentsByClass(cls.Id);
+       System.assertEquals('xyz', namesOfStudents.get(0).Name, 'No student by such name is present in the class');
+   }
+    
+    /**
+    * Test method to get the student name in a particular class.
+    */ 
+   @isTest
+   public static void getNamesOfMultipleStudent()
+   {
+       Class__c cls = new Class__c(Name='x',MaxSize__c=2);
+       insert cls;
+       List<Student__c> studentsInClass = new List<Student__c>();
+       for(Integer i = 0; i < 101; i++)
```

Comment:
Class max size is 2 how is it inserting 101 students?

Reply:
The trigger of Max size was not working correctly at that time.
That's why this test case was working.
Now everything is working as expected.